src/neuroglancer/datasource/graphene/frontend.ts

```diff
@@ -702,3 +1011,313 @@ class SliceViewPanelChunkedGraphLayer extends SliceViewPanelRenderLayer {
     }
   }
 }
+
+const ANNOTATE_MULTICUT_SEGMENTS_TOOL_ID = 'multicutSegments';
+
+class MulticutAnnotationLayerView extends AnnotationLayerView {
+  private _annotationStates: MergedAnnotationStates;
+
+  constructor(
+      public layer: Borrowed<UserLayerWithAnnotations>,
+      public displayState: AnnotationDisplayState) {
+    super(layer, displayState);
+  }
+
+  get annotationStates() {
+    if (this._annotationStates === undefined) {
+      this._annotationStates = this.registerDisposer(new MergedAnnotationStates());
+    }
+    return this._annotationStates;
+  }
+}
+
+const synchronizeAnnotationSource = (source: WatchableSet<SegmentSelection>, state: AnnotationLayerState) => {
+  const annotationSource = state.source;
+
+  annotationSource.childDeleted.add(annotationId => {
+    const selection = [...source].find(selection => selection.annotationReference?.id === annotationId)
+    if (selection) source.delete(selection); 
+  });
+
+  const addSelection = (selection: SegmentSelection) => {
+    const annotation: Point = {
+      id: '',
+      point: selection.position,
+      type: AnnotationType.POINT,
+      properties: [],
+      relatedSegments: [[selection.segmentId, selection.rootId]],
+    };
+    const ref = annotationSource.add(annotation);
+    selection.annotationReference = ref;
+  }
+
+  source.changed.add((x, add) => {
+    if (x === null) {
+      for (const annotation of annotationSource) {
+        // using .clear does not remove annotations from the list
+        // (this.blueGroupAnnotationState.source as LocalAnnotationSource).clear();
+        annotationSource.delete(annotationSource.getReference(annotation.id));
+      }
+      return;
+    }
+
+    if (add) {
+      addSelection(x);
+    } else if (x.annotationReference) {
+      annotationSource.delete(x.annotationReference);
+    }
+  });
+
+  // load initial state
+  for (const selection of source) {
+    addSelection(selection);
+  }
+}
+
+export class GrapheneTab extends Tab {
+  private annotationLayerView =
+      this.registerDisposer(new MulticutAnnotationLayerView(this.layer, this.layer.annotationDisplayState));
+
+  constructor(public layer: SegmentationUserLayer) {
+    super();
+    const {graphConnection} = layer;
+    if (graphConnection instanceof GraphConnection) {
+      const states = graphConnection.initializeAnnotations(this.layer);
```

Comment:
I think the intention is that the multicut markers are shown all the time if present, regardless of whether the annotation tab is open or the multicut tool has been activated.  If so, then `initializeAnnotations` should instead happen earlier, e.g. when the graph connection is created.

Reply:
Yes, I think adding a `layer` paremeter to `SegmentationGraphSource.connect` would be reasonable.

I'm not too fond of making subsources required, as I'd rather preserve user control over that ---- but I think it is reasonable if the meshes don't work with the graph subsource disabled.  Potentially there could be a way to specify that a subsource depends on another subsource, e.g. to prevent the mesh subsource from being enabled without the graph subsource.  But I think that may not be worth the trouble to implement.

Reply:
does https://github.com/google/neuroglancer/pull/395/commits/b16feb6df1ac27844f6a2fa1102e02fe96073ef4 seem reasonable? I replaced segmentsState with layer since you can access the segmentState from the layer and having them as two separate arguments opens it up to sending unconnected arguments. Also, transform was accidentally added in the graphene view PR so I removed that.

Reply:
Yes, thanks.